VirtualH89/Src/MemoryDecoder.h

```diff
@@ -0,0 +1,55 @@
+///
+/// \file MemoryDecoder.h
+///
+/// Base class for memory decoders, which implement various layouts of
+/// the address space along with switching between them.
+///
+/// \date Mar 05, 2016
+/// \author Douglas Miller
+///
+#ifndef MEMORYDECODER_H_
+#define MEMORYDECODER_H_
+
+#include "Memory8K.h"
+#include "MemoryLayout.h"
+#include "GppListener.h"
+
+class MemoryDecoder: public GppListener
+{
```

Comment:
My thinking was that on H89s (H90s) there are GPP bits that feed into the memory decoder chips, so I thought it reasonable to make all (H89) memory decoder instances derive from gppListener. I'm not terribly attached to that idea, though.